app/lib/methods/sendMessage.ts

```diff
@@ -9,6 +9,42 @@ import { E2EType, IMessage, IUser, TMessageModel } from '../../definitions';
 import sdk from '../services/sdk';
 import { E2E_MESSAGE_TYPE, E2E_STATUS, messagesStatus } from '../constants';
 
+export async function resendFailedMessages(): Promise<void> {
+	try {
+		const db = database.active;
+		const messageRecords = await db
+			.get('messages')
+			.query(
+				Q.or(Q.where('status', messagesStatus.ERROR), Q.where('status', messagesStatus.TEMP)),
+				Q.where('tmid', null),
+				Q.experimentalSortBy('ts', Q.asc)
+			)
+			.fetch();
+
+		const threadRecords = await db
+			.get('thread_messages')
+			.query(
+				Q.or(Q.where('status', messagesStatus.ERROR), Q.where('status', messagesStatus.TEMP)),
+				Q.experimentalSortBy('ts', Q.asc)
+			)
+			.fetch();
+
+		const resendMsgs: Promise<void>[] = [];
+
+		messageRecords.forEach((message: TMessageModel) => {
+			resendMsgs.push(resendMessage(message));
+		});
+
+		threadRecords.forEach((message: TMessageModel) => {
+			resendMsgs.push(resendMessage(message, message.rid));
+		});
```

Comment:
I'm wondering if we should unify these arrays and sort by ts again, to make sure messages are in order.
Can you test?

Reply:
@diegolmello Yes you're correct that would be safer option. but in the real scenario the probability is very low as people have decent internet connection and no one would send many msgs at once